### DISCUSSION TITLE
In basic_chain dc power only for only single module, rather than for array of multiple modules?

### DISCUSSION THREAD

**Discussion by Kaesekopf** [pvlib/pvlib-python#1385] [Q&A] [answered]
I am just starting with python pvlib and the basic_chain function seemed a good way to get acquainted . However, one thing really wonders me. The PV-array is only described with the module_parameters argument, and there is no parameter or keyword to describe the size of the array.

Also the code suggests that only the dc- power of a single module is calculated, and when I run the function for a 250 W module and a 5kW inverter the AC output is never comes above 200W, again confirming by suspicion that only the DC output of a single module is calculated.

As a quick workaround I have made  a local version of the basic_chain function with an array-inverter power ratio as additional output and a scaling of the DC output within the function*. This  solved my problem for the moment, but I still have some questions:

1.     Am I overlooking something?
2.     Is this an intended feature, for example, the function is only intended to be used for so-called AC-modules (a single module connected to it's own micro-inverter). If so shouldn't this be in the documentation?
3.     If my suspicion is right, should this be an official issue?

*The changed code:

```
    #match array STC power with that of inverter
    P_STC_array=array_inverter_ratio*inverter_parameters['Pdco']
    nrofmodules=P_STC_array/(module_parameters['Impo']*module_parameters['Vmpo'])
    
    #determine number of module in series by matching Voltage
    nrofmodules_in_series=inverter_parameters['Vdco']/module_parameters['Vmpo']
   
    #scale P and V of single module to that of full PV-array
    ac = pvlib.inverter.sandia(nrofmodules_in_series*dc['v_mp'], nrofmodules*dc['p_mp'], inverter_parameters)
```

"array_inverter_ratio" is a new keyword with default value 1.
This is a bit crude, because it allows a non-integer number of modules, and a default None value could be introduced for the keyword to make the function backwards compatible with its' current behaviour, but for the moment it is good enough for my purposes.

**Comment by kandersolar** [comment, marked as answer]
Hi @Kaesekopf, thanks for your interest in pvlib!  I confess I have never used `basic_chain` and was only vaguely aware that it existed.  Skimming the commits in https://github.com/pvlib/pvlib-python/pull/93, I get the sense that it was something of a prototype for the eventual `ModelChain` class.  Anyway, instead of modifying `basic_chain` to suit your needs, I suggest using the more detailed modeling workflow provided by `ModelChain`, `PVSystem`, and `Array`.  The docs have some pages and gallery examples to get started there.  I'll point out that `Array` has parameters for the number of modules per string and the total number of strings :)

I do wonder if we should add a short note to its docstring explaining the limitations and pointing users to the more capable alternative.

**Comment by Kaesekopf** [comment]
Thanks. 
I came to this issue, because basic_chain is used in feedinlib , which in its' turn is used in another project that I contribute to.

**Comment by cwhanse** [comment]
@Kaesekopf I think feedinlib is using `modelchain.ModelChain`, rather than `basic_chain`. See [here](https://github.com/oemof/feedinlib/blob/b4d0481e74100a33a397c23e6bae531f0f4cbada/src/feedinlib/models/pvlib.py#L18).

@kanderso-nrel I think you are right, that `basic_chain` was a prototype for `ModelChain`. Looking at it now, I think it is troublesome. For example, it hardwires the SAPM module model, and there are no parameters for most modules (the last addition to that parameter file was about 10 years ago). Personally, I think `basic_chain` should be retired, and users directed to `ModelChain.with_pvwatts`. There is at [least one comment](https://github.com/oemof/feedinlib/issues/62#issuecomment-966078208) on a feedinlib issue suggesting that they abandon pvlib because of concerns about pvlib requiring too many inputs.
